internal/cache/inmemory.go

```diff
@@ -51,9 +60,6 @@ func (i *inmemory) Set(k string, res types.DataResponseV1) error {
 		return err
 	}
 
-	if err := i.tc.Set(k, v); err != nil {
-		return fmt.Errorf("failed to store in in-memory cache: %w", err)
-	}
-
+	i.tc.Set(k, v, ttlcache.DefaultTTL)
```

Comment:
nit. I suggest to add a comment for the reader here on what `ttlcache.DefaultTTL` means. On the first pass I read this as "use a package default TTL and overwrite our configuration`. However the `ttlcache` package means with `DefaultTTL` something different. I guess they have used a ambiguous name here.